benchmarks/benchmarks/bench_io.py

```diff
@@ -62,3 +63,181 @@ def setup(self):
+                          ('category_4', 'S1'),
+                          ('category_5', 'f8')])
+        self.csv_data.seek(0)
+
+
+class LoadtxtCSVSkipRows(Benchmark):
+    # benchmarks for loadtxt row skipping when
+    # reading in csv file data; a similar benchmark
+    # is present in the pandas asv suite
+
+    params = [0, 500, 10000]
+    param_names = ['skiprows']
+
+    def setup(self, skiprows):
+        np.random.seed(123)
+        test_array = np.random.rand(100000, 3)
+        self.fname = 'test_array.csv'
+        np.savetxt(fname=self.fname,
+                   X=test_array,
+                   delimiter=',')
+
+    def time_skiprows_csv(self, skiprows):
+        np.loadtxt(self.fname,
+                   delimiter=',',
+                   skiprows=skiprows)
+
+class LoadtxtReadUint64Integers(Benchmark):
+    # pandas has a similar CSV reading benchmark
+    # modified to suit np.loadtxt
+
+    params = [550, 1000, 10000]
+    param_names = ['size']
+
+    def setup(self, size):
+        arr = np.arange(size).astype('uint64') + 2**63
+        self.data1 = StringIO(u'\n'.join(arr.astype(str).tolist()))
+        arr = arr.astype(object)
+        arr[500] = -1
+        self.data2 = StringIO(u'\n'.join(arr.astype(str).tolist()))
+
+    def time_read_uint64(self, size):
+        # mandatory rewind of StringIO object
+        # between iterations of a repeat:
+        np.loadtxt(self.data1)
+        self.data1.seek(0)
+
+    def time_read_uint64_neg_values(self, size):
+        # mandatory rewind of StringIO object
+        # between iterations of a repeat:
+        np.loadtxt(self.data2)
+        self.data2.seek(0)
+
+class LoadtxtUseColsCSV(Benchmark):
+    # benchmark selective column reading from CSV files
+    # using np.loadtxt
+
+    params = [2, [1, 3], [1, 3, 5, 7]]
+    param_names = ['usecols']
+
+    def setup(self, usecols):
+        num_lines = 5000
+        data = [u'0, 1, 2, 3, 4, 5, 6, 7, 8, 9'] * num_lines
+        self.csv_data = StringIO(u'\n'.join(data))
+
+    def time_loadtxt_usecols_csv(self, usecols):
+        # must rewind StringIO because of state
+        # dependence of file reading
+        np.loadtxt(self.csv_data,
+                   delimiter=u',',
+                   usecols=usecols)
+        self.csv_data.seek(0)
+
+class LoadtxtCSVDateTime(Benchmark):
+    # benchmarks for np.loadtxt operating with
+    # datetime data in a CSV file
+
+    params = [20, 200, 2000, 20000]
+    param_names = ['num_lines']
+
+    def setup(self, num_lines):
+        # create the equivalent of a two-column CSV file
+        # with date strings in the first column and random
+        # floating point data in the second column
+        dates = np.arange('today', 20, dtype=np.datetime64)
+        np.random.seed(123)
+        values = np.random.rand(20)
+        date_line = u''
+        index = 0
+
+        for entry in dates:
+            date_line += (str(entry) + ',' + str(values[index]) + '\n')
```

Comment:
There's some silent `str` -> `unicode` conversion happening here on python2, but I suppose all the strings are ascii anyway